samples/sensor/qdec/sample.yaml

```diff
@@ -8,7 +8,12 @@ common:
 
 tests:
   sample.sensor.qdec_sensor:
-    filter: dt_alias_exists("qdec0")
+    platform_allow:
```

Comment:
`platform_allow` makes it impossible for twister to exercise this sample on an out-of-tree board with a qdec sensor.

Reply:
and why do want to do that? we are verifying samples here, this is not a test and the sample should be verified on the document and available boards in the tree with this sensor. Relying on the filter means that I have to run cmake > 800 times  just to see which board has qdec0 in their dts

something like that...

```
i9:zephyr(main): ./scripts/twister -T samples/sensor/qdec --all -v
Renaming output directory to /home/nashif/zephyrproject/zephyr/twister-out.2
INFO    - Using Ninja..
INFO    - Zephyr version: v3.7.0-4713-g091c6664c51b
INFO    - Using 'zephyr' toolchain.
INFO    - Selecting all possible platforms per test case
INFO    - Building initial testsuite list...
INFO    - Writing JSON report /home/nashif/zephyrproject/zephyr/twister-out/testplan.json
INFO    - JOBS: 20
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - 2592/3420 riscv32_virtual           samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2593/3420 intel_adl_rvp             samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2594/3420 intel_adl_crb             samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2595/3420 ek_ra4m2                  samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2596/3420 nucleo_wb09ke             samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2597/3420 frdm_ke17z512             samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2598/3420 nrf54l15dk/nrf54l15/cpuflpr samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2599/3420 nrf54l15dk/nrf54l15/cpuflpr/xip samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2600/3420 robokit1                  samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2601/3420 steval_fcu001v1           samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2602/3420 mg100                     samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2603/3420 olimex_stm32_e407         samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2604/3420 olimex_esp32_evb/esp32/appcpu samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2605/3420 esp32c3_devkitm           samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2606/3420 stm32l152c_disco          samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2607/3420 stm32l1_disco             samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2608/3420 titanium_ti60_f225        samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2609/3420 olimex_esp32_evb/esp32/procpu samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2610/3420 esp32s3_devkitm/esp32s3/appcpu samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2611/3420 arduino_due               samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2612/3420 b_g474e_dpow1             samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2613/3420 arduino_portenta_h7/stm32h747xx/m4 samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2614/3420 arduino_portenta_h7/stm32h747xx/m7 samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2615/3420 em_starterkit@2.2/emsk_em7d samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2616/3420 em_starterkit/emsk_em7d   samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2617/3420 gd32e507v_start           samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2618/3420 em_starterkit/emsk_em9d   samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2619/3420 em_starterkit/emsk_em11d  samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2620/3420 stm32g0316_disco          samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2621/3420 innblue22/nrf9160         samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2622/3420 mini_stm32h743            samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2623/3420 innblue22/nrf9160/ns      samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2624/3420 slwrb4161a                samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2625/3420 stm32l476g_disco          samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2626/3420 fvp_base_revc_2xaemv8a//smp/ns samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2627/3420 fvp_base_revc_2xaemv8a    samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2628/3420 bt510                     samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2629/3420 olimex_lora_stm32wl_devkit samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2630/3420 olimex_lora_stm32wl_devkit@D samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2631/3420 nucleo_g070rb             samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2632/3420 cc1352r1_launchxl         samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2633/3420 max32670evkit             samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2634/3420 it8xxx2_evb               samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2635/3420 nucleo_f412zg             samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2636/3420 pico_pi/mcimx7d/m4        samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2637/3420 esp32_ethernet_kit/esp32/procpu samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2638/3420 esp32_ethernet_kit/esp32/appcpu samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2639/3420 xiao_ble                  samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2640/3420 slwrb4180a                samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2641/3420 xiao_ble/nrf52840/sense   samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2642/3420 stm32h745i_disco/stm32h745xx/m4 samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2643/3420 stm32h745i_disco/stm32h745xx/m7 samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2644/3420 max32690fthr/max32690/m4  samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2645/3420 stm32f401_mini            samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2646/3420 nucleo_g0b1re             samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 2647/3420 96b_meerkat96/mcimx7d/m4  samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
```


and just by using default twister options, i.e. what we have in CI:

```
i9:zephyr(main): ./scripts/twister -s samples/sensor/qdec/sample.sensor.qdec_sensor   -v
Renaming output directory to /home/nashif/zephyrproject/zephyr/twister-out.4
INFO    - Using Ninja..
INFO    - Zephyr version: v3.7.0-4713-g091c6664c51b
INFO    - Using 'zephyr' toolchain.
INFO    - Selecting default platforms per test case
INFO    - Building initial testsuite list...
INFO    - Writing JSON report /home/nashif/zephyrproject/zephyr/twister-out/testplan.json
INFO    - JOBS: 20
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - 16/44 qemu_arc/qemu_arc_hs5x    samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 17/44 qemu_arc/qemu_arc_hs/xip  samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 18/44 qemu_cortex_a9            samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 19/44 qemu_leon3                samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 20/44 qemu_malta                samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 21/44 qemu_malta/qemu_malta/be  samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 22/44 qemu_arc/qemu_arc_hs6x    samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 23/44 qemu_xtensa/dc233c        samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 24/44 mps2/an385                samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 25/44 qemu_riscv32              samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 26/44 qemu_arc/qemu_arc_hs      samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 27/44 m2gl025_miv               samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 28/44 qemu_riscv32e             samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 29/44 qemu_nios2                samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 30/44 qemu_riscv64/qemu_virt_riscv64/smp samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 31/44 qemu_riscv32/qemu_virt_riscv32/smp samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 32/44 mps2/an521/cpu0           samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 33/44 qemu_xtensa/dc233c/mmu    samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 34/44 qemu_riscv64              samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 35/44 qemu_arc/qemu_arc_em      samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 36/44 nsim/nsim_hs/mpuv6        samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 37/44 qemu_cortex_a53           samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 38/44 qemu_x86                  samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 39/44 qemu_cortex_r5            samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 40/44 qemu_cortex_a53/qemu_cortex_a53/smp samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 41/44 qemu_cortex_m0            samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 42/44 native_sim                samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 43/44 qemu_x86_64               samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 44/44 nsim/nsim_sem             samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)

INFO    - 1 test scenarios (44 test instances) selected, 44 configurations skipped (15 by static filter, 29 at runtime).
INFO    - 0 of 44 test configurations passed (0.00%), 0 built (not run), 0 failed, 0 errored, 44 skipped with 0 warnings in 10.07 seconds
INFO    - In total 0 test cases were executed, 44 skipped on 44 out of total 856 platforms (5.14%)
INFO    - 0 test configurations executed on platforms, 0 test configurations were only built.
INFO    - Saving reports...
INFO    - Writing JSON report /home/nashif/zephyrproject/zephyr/twister-out/twister.json
INFO    - Writing xunit report /home/nashif/zephyrproject/zephyr/twister-out/twister.xml...
INFO    - Writing xunit report /home/nashif/zephyrproject/zephyr/twister-out/twister_report.xml...
INFO    - Run completed

```


NOTHING get built, so we have been wasting time and resourcing running cmake on this "test" 100s of times without actually verifying anything, defeats the purpose completely.

With this change:


```
i9:zephyr(topic/samples/filters_3): ./scripts/twister -s samples/sensor/qdec/sample.sensor.qdec_sensor   -v
Renaming output directory to /home/nashif/zephyrproject/zephyr/twister-out.5
INFO    - Using Ninja..
INFO    - Zephyr version: v3.7.0-4718-g3066e108c138
INFO    - Using 'zephyr' toolchain.
INFO    - Selecting default platforms per test case
INFO    - Building initial testsuite list...
INFO    - Writing JSON report /home/nashif/zephyrproject/zephyr/twister-out/testplan.json
INFO    - JOBS: 20
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - 45/45 mimxrt1050_evk            samples/sensor/qdec/sample.sensor.qdec_sensor      NOT RUN (build)

INFO    - 1 test scenarios (45 test instances) selected, 44 configurations skipped (44 by static filter, 0 at runtime).
INFO    - 0 of 45 test configurations passed (0.00%), 1 built (not run), 0 failed, 0 errored, 44 skipped with 0 warnings in 11.41 seconds
INFO    - In total 0 test cases were executed, 44 skipped on 45 out of total 856 platforms (5.26%)
INFO    - 0 test configurations executed on platforms, 1 test configurations were only built.
INFO    - Saving reports...
INFO    - Writing JSON report /home/nashif/zephyrproject/zephyr/twister-out/twister.json
INFO    - Writing xunit report /home/nashif/zephyrproject/zephyr/twister-out/twister.xml...
INFO    - Writing xunit report /home/nashif/zephyrproject/zephyr/twister-out/twister_report.xml...
INFO    - Run completed
```


if I remove the platform_allow in this change and restore the filter:

```
i9:zephyr(topic/samples/filters_3): ./scripts/twister -s samples/sensor/qdec/sample.sensor.qdec_sensor   -v
Renaming output directory to /home/nashif/zephyrproject/zephyr/twister-out.6
INFO    - Using Ninja..
INFO    - Zephyr version: v3.7.0-4718-g3066e108c138
INFO    - Using 'zephyr' toolchain.
INFO    - Selecting default platforms per test case
INFO    - Building initial testsuite list...
INFO    - Writing JSON report /home/nashif/zephyrproject/zephyr/twister-out/testplan.json
INFO    - JOBS: 20
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - 16/45 nsim/nsim_hs/mpuv6        samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 17/45 qemu_riscv64/qemu_virt_riscv64/smp samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 18/45 qemu_xtensa/dc233c/mmu    samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 19/45 qemu_leon3                samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 20/45 qemu_malta/qemu_malta/be  samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 21/45 qemu_xtensa/dc233c        samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 22/45 m2gl025_miv               samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 23/45 nsim/nsim_sem             samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 24/45 mps2/an521/cpu0           samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 25/45 qemu_cortex_a53           samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 26/45 qemu_cortex_a53/qemu_cortex_a53/smp samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 27/45 qemu_x86_64               samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 28/45 qemu_nios2                samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 29/45 qemu_x86                  samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 30/45 mps2/an385                samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 31/45 qemu_riscv64              samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 32/45 qemu_riscv32/qemu_virt_riscv32/smp samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 33/45 qemu_malta                samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 34/45 qemu_cortex_m0            samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 35/45 qemu_arc/qemu_arc_hs5x    samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 36/45 qemu_riscv32              samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 37/45 qemu_cortex_r5            samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 38/45 native_sim                samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 39/45 qemu_cortex_a9            samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 40/45 qemu_arc/qemu_arc_em      samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 41/45 qemu_riscv32e             samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 42/45 qemu_arc/qemu_arc_hs6x    samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 43/45 qemu_arc/qemu_arc_hs/xip  samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 44/45 qemu_arc/qemu_arc_hs      samples/sensor/qdec/sample.sensor.qdec_sensor      SKIPPED (runtime filter)
INFO    - 45/45 mimxrt1050_evk            samples/sensor/qdec/sample.sensor.qdec_sensor      NOT RUN (build)

INFO    - 1 test scenarios (45 test instances) selected, 44 configurations skipped (15 by static filter, 29 at runtime).
INFO    - 0 of 45 test configurations passed (0.00%), 1 built (not run), 0 failed, 0 errored, 44 skipped with 0 warnings in 17.37 seconds
INFO    - In total 0 test cases were executed, 44 skipped on 45 out of total 856 platforms (5.26%)
INFO    - 0 test configurations executed on platforms, 1 test configurations were only built.
INFO    - Saving reports...
INFO    - Writing JSON report /home/nashif/zephyrproject/zephyr/twister-out/twister.json
INFO    - Writing xunit report /home/nashif/zephyrproject/zephyr/twister-out/twister.xml...
INFO    - Writing xunit report /home/nashif/zephyrproject/zephyr/twister-out/twister_report.xml...
INFO    - Run completed

```

same thing, we now at least build on one, but still waste time on trying to figure out who has this sensor.


and this is one of the reasons why our CI is very slow, and this is just the tip of the iceberg.

We are looking for way to improve the logic of inclusion/exclusions in tests/sample via twister but the above and many other samples contribute to a large portion why we spend so much time building in CI.

Reply:
Why can't we use `depends on/supports` yaml properties?